lib/state_change_syncer.go

```diff
@@ -174,8 +175,11 @@ func (stateChangeEntry *StateChangeEntry) RawDecodeWithoutMetadata(blockHeight u
 	ancestralRecord := stateChangeEntry.EncoderType.New()
 	if exist, err := DecodeFromBytes(ancestralRecord, rr); exist && err == nil {
 		stateChangeEntry.AncestralRecord = ancestralRecord
+		stateChangeEntry.AncestralRecordBytes = EncodeToBytes(blockHeight, ancestralRecord)
 	} else if err != nil {
 		return errors.Wrapf(err, "StateChangeEntry.RawDecodeWithoutMetadata: error decoding ancestral record")
+	} else {
+		stateChangeEntry.AncestralRecordBytes = EncodeToBytes(blockHeight, nil)
```

Comment:
so I think I get what's going on here? for the `else` clause, we just want to encode an empty struct which signifies that the entry didn't exist before. and in the `if exists && err == nil` case, we get the bytes while we can. Can we add a small comment around this section just to clarify?

Reply:
Exactly - will do